lib/workflows/utils.js

```diff
@@ -34,16 +34,35 @@ const check = (config) => {
     return 'Workflow steps must be an array';
   }
 
-  config.steps.forEach(step => {
-    if (!step.name) {
+  if (config.steps.length < 1) {
+    return 'Workflow must have at least one step';
+  }
+
+  const test = config.steps.map(step => {
+    if (!step.name || step.name === '') {
       return 'Step must have a name';
     }
+    if (typeof step.name !== 'string') {
+      return 'Step name must be a string';
+    }
     if (step.self && typeof step.self !== 'boolean') {
-      return 'Whether a step is self-publish must be a boolean';
+      return 'Self-publish must be a boolean';
+    }
+
+    return true;
+  }).filter((res) => {
+    if (res !== true) {
+      return res;
```

Comment:
Filter should return `true` or `false`, not a truthy thing like a string. Filter does not manipulate its array's value, only determines if it should be in the array.